Entities/Common/Building/BlobPlacement.as

```diff
@@ -346,10 +355,23 @@ void onTick(CBlob@ this)
 			{
 				if (snap && bc.cursorClose && bc.hasReqs && bc.buildable && bc.supported)
 				{
-					CBitStream params;
-					params.write_u16(carryBlob.getNetworkID());
-					params.write_Vec2f(getBottomOfCursor(bc.tileAimPos, carryBlob));
-					this.SendCommand(this.getCommandID("placeBlob"), params);
+					CMap@ map = getMap();
+					CBlob@ blobAtPos = map.getBlobAtPosition(getBottomOfCursor(bc.tileAimPos, carryBlob));
+					if (blobAtPos !is null && carryBlob.getName() == blobAtPos.getName() && blobAtPos.getHealth() < blobAtPos.getInitialHealth() && blobAtPos.getName() != "ladder")
+					{
+						CBitStream params;
+						params.write_u16(carryBlob.getNetworkID());
+						params.write_Vec2f(getBottomOfCursor(bc.tileAimPos, carryBlob));
+						this.SendCommand(this.getCommandID("repairBlob"), params);
+					}
+					else
+					{
+						CBitStream params;
+						params.write_u16(carryBlob.getNetworkID());
+						params.write_Vec2f(getBottomOfCursor(bc.tileAimPos, carryBlob));
+						this.SendCommand(this.getCommandID("placeBlob"), params);
+					}
```

Comment:
could write to params before the if statement, and just send a different cmd based on the statement, less code dupe

Reply:
fixed